addons/sale_product_configurator/static/src/js/product_configurator_widget.js

```diff
@@ -230,17 +212,24 @@ var ProductConfiguratorWidget = relationalFields.FieldMany2One.extend({
             preventProductIdCheck: true,
             changes: this._getMainProductChanges(result.mainProduct),
             onSuccess: function () {
-                if (result.options) {
-                    var parentList = self.getParent();
-                    self.trigger_up('add_record', {
-                        context: self._productsToRecords(result.options),
-                        forceEditable: 'bottom',
-                        allowWarning: true,
-                        onSuccess: function (){
+                var parentList = self.getParent();
+                self._onProductChange(result.mainProduct.product_id, dataPointId).then(function (res) {
+                    if (result.options && result.options.length !== 0) {
+
+                      // VFE TODO should be given to product change for the options to be applied after _onProductChange is totally ended.
+                      self.trigger_up('add_record', {
+                          context: self._productsToRecords(result.options),
+                          forceEditable: 'bottom',
+                          allowWarning: true,
+                          onSuccess: function () {
                             parentList.unselectRow();
-                        }
-                    });
-                }
+                          }
+                      });
+
+                    } else if (!res) {
```

Comment:
Don't we always want to do that if there are no options?
Or maybe not if we need to "configure" something else first?

This probably deserves a little comment.

Reply:
Also seems weird that the same line in the `onSuccess` of the `add_record` doesn't care about "res".

Reply:
The first unselectRow is applied on the last options row added and can be always called after the options have been added.

The second one applies to the row of the product being actually configured and shouldn't be called if res=true which means that a _onProductChange configurator has been opened (and unselecting the row will have to be done after the changes of this configurator).

Reply:
Well this is why calling that `stopPropagation` or something like that would help to understand the flow imo :)

And even with that it would maybe be nice if that part was extracted to another method with some nice doc to be able to still understand what it does in the future.